internal/tsoptions/errors.go

```diff
@@ -67,3 +84,16 @@ func createDiagnosticForNodeInSourceFileOrCompilerDiagnostic(sourceFile *ast.Sou
 	}
 	return ast.NewCompilerDiagnostic(message, args...)
 }
+
+func extraKeyDiagnostics(s string) *diagnostics.Message {
```

Comment:
sorry i meant here shouldnt this be from parser.unknownDiagnostics ? i dont understand why we have two ways of doing same thing

Reply:
They are implemented in two different ways because of how it was implemented in strada, and then the two different code paths that use the messages.
When we are parsing from a `map` to a `struct`, we only have an `optionParser`, we don't know what type of `optionParser` it is, so we can't use `extraKeyDiagnostics` directly. 
When we are parsing the tsconfig, we don't use `optionParser`, so we are not able call `optionsParser.UnknownOptionDiagnostic()`.

I could refactor each of the `optionsParser.UnknownOptionDiagnostic()` to be like 
```
func (o *typeAcquisitionParser) UnknownOptionDiagnostic() *diagnostics.Message {
	return extraKeyDiagnostics("typeAcquisition")
}
```
but I'm not sure if it's worth the indirection

Reply:
I thought about it some more, and ended up doing that refactor